test/unit/core.js

```diff
@@ -791,6 +791,17 @@ QUnit.test( "jQuery.map", function( assert ) {
 		return k % 2 ? k : [ k, k, k ];
 	} );
 	assert.equal( result.join( "" ), "00012223", "Array results flattened (#2616)" );
+
+	result = jQuery.map( [ [ [ 1, 2 ], 3 ], 4 ], function( v, k ) {
+		return v;
+	} );
+	assert.equal( result.length, 3, "Array flatten only one level down" );
+	assert.ok( Array.isArray( result[ 0 ] ), "Array flatten only one level down" );
+
+	result = jQuery.map( Array( 300000 ), function( v, k ) {
+		return k;
+	} );
+	assert.equal( result.length, 300000, "Able to map 300000 records without any problems (#4320)" );
```

Comment:
As I understand, this will fail in browsers without `Array#flat` support. So we need to guard those tests with a check for `Array.prototype.flat`. Other browsers need dummy assertions so that the count passed to `assert.expect` matches. See an example:
https://github.com/jquery/jquery/blob/b334ce7735ae453bd5643b251f36c3cedce4b3de/test/unit/selector.js#L24-L30